### DISCUSSION TITLE
Error with Engine testing: ValueError: Encountered two metrics both named AUROC

### DISCUSSION THREAD

**Discussion by yhtanc** [open-edge-platform/anomalib#1948] [Q&A]
I am trying to run FastFlow model via Anomalib API. But I constantly get this error.
Is there something that I am missing on the code?

    engine.test(datamodule=datamodule, model=model)

  File ~\anomalib\src\anomalib\engine\engine.py:662 in test
    return self.trainer.test(model, dataloaders, ckpt_path, verbose, datamodule)

  File ~\anaconda3\envs\anomalib_env\lib\site-packages\lightning\pytorch\trainer\trainer.py:754 in test
    return call._call_and_handle_interrupt(

  File ~\anaconda3\envs\anomalib_env\lib\site-packages\lightning\pytorch\trainer\call.py:44 in _call_and_handle_interrupt
    return trainer_fn(*args, **kwargs)

  File ~\anaconda3\envs\anomalib_env\lib\site-packages\lightning\pytorch\trainer\trainer.py:794 in _test_impl
    results = self._run(model, ckpt_path=ckpt_path)

  File ~\anaconda3\envs\anomalib_env\lib\site-packages\lightning\pytorch\trainer\trainer.py:950 in _run
    call._call_setup_hook(self)  # allow user to setup lightning_module in accelerator environment

  File ~\anaconda3\envs\anomalib_env\lib\site-packages\lightning\pytorch\trainer\call.py:93 in _call_setup_hook
    _call_callback_hooks(trainer, "setup", stage=fn)

  File ~\anaconda3\envs\anomalib_env\lib\site-packages\lightning\pytorch\trainer\call.py:208 in _call_callback_hooks
    fn(trainer, trainer.lightning_module, *args, **kwargs)

  File ~\anomalib\src\anomalib\callbacks\metrics.py:102 in setup
    pl_module.pixel_metrics.add_metrics(new_metrics[name])

  File ~\anaconda3\envs\anomalib_env\lib\site-packages\torchmetrics\collections.py:361 in add_metrics
    raise ValueError(f"Encountered two metrics both named {name}")

ValueError: Encountered two metrics both named AUROC

**Comment by samet-akcay** [comment]
Hi @yhtanc, thanks for reporting this. Can you provide your full code so we could reproduce?

**Comment by yhtanc** [comment]
I just adapted the "201_fastflow.ipynb" code to play a little bit the with
Spyder. I already created an environment
as told on github. I attached the code. I got the error when I try to fit
the engine.

Samet Akcay ***@***.***>, 4 Nis 2024 Per, 10:56 tarihinde
şunu yazdı:
